ops/model.py

```diff
@@ -469,12 +469,16 @@ class Unit:
     app: Application
     """Application the unit is part of."""
 
+    number: int
+    """Unit ID number, for example 0."""
+
     def __init__(self, name: str, meta: 'ops.charm.CharmMeta',
                  backend: '_ModelBackend', cache: '_ModelCache'):
         self.name = name
 
         app_name = name.split('/')[0]
         self.app = cache.get(Application, app_name)
+        self.number = int(name.split('/')[1])
```

Comment:
```suggestion
        app_name, self.ordinal = name.split('/')
        self.app = cache.get(Application, app_name)
```

A slight simplification here, perhaps? I wonder if `ordinal` might be a better name for this?

Reply:
Yeh, fair point, I think it would be a string.

Perhaps

```python
app_name, ordinal = name.split('/')
self.app = cache.get(Application, app_name)
self.ordinal = int(ordinal)
```

Reply:
Would need to update docstrings to use the new term "ordinal" (which I imagine is better than the existing "number", "id").